src/blockchain/compact_filters/nakamoto.rs

```diff
@@ -0,0 +1,677 @@
+                            debug!(
+                                "Processing Confirmed Transaction : {} at Block : {}",
+                                tx.txid(),
+                                height
+                            );
+                            add_tx(
+                                database,
+                                &tx,
+                                Some(height as u32),
+                                Some(timestamp as u64),
+                                &mut internal_max_deriv,
+                                &mut external_max_deriv,
+                            )?;
+                        }
+                    }
+                }
+                Event::FeeEstimated { height, fees, .. } => {
+                    self.add_fee_data(height as u32, fees);
+                }
+                Event::TxStatusChanged { txid, status } => {
+                    match status {
+                        TxStatus::Unconfirmed => {
+                            debug!("txid:{}, status : Broadcasted", txid);
+                        }
+                        TxStatus::Acknowledged { peer } => {
+                            debug!("txid: {}, status : ACK by {}", txid, peer);
+                        }
+                        TxStatus::Confirmed { height, .. } => {
+                            debug!("txid: {}, status : Confirmed at {}", txid, height);
+                        }
+                        TxStatus::Reverted => {
+                            debug!("Transaction reverted due to reorg: txid : {}", txid);
+                            let _ = unconfirm_tx(database, &txid)?;
+                        }
+                        TxStatus::Stale { replaced_by, block } => {
+                            debug!(
+                                "txid: {}, status : Replaced by : txid {} at blockhash {}",
+                                txid, replaced_by, block
+                            );
+                            let tx = database.get_tx(&replaced_by, true)?.expect(
+                                "Wallet transaction got replaced by unknown foreign transaction",
+                            ); // TODO: Handle the case when we don't have the overriding transaction
+                            delete_tx(database, &tx.txid)?;
+                        }
+                    }
+                }
+                Event::Synced { height, tip } => {
+                    debug!("Sync Status : {}:{}", height, tip);
+                    progress_update.update(
+                        (height as f32 / tip as f32) * 100.0,
+                        Some("Sync Progress".into()),
+                    )?;
+                    // We break the sync loop once we reach current height.
+                    if height == tip {
+                        debug!("Sync complete at : {}", height);
+                        self.last_sync_height.set(height as u32);
+                        break;
+                    }
```

Comment:
I assume you have another mechanism to restart the event loop when a new block header is received.

Reply:
No, this is basically a hack to stop the sync loop for the reorg tests in regetst.. Ideally an `invalidateblock` call on the regtest should also update the tip height, and the sync should automatically stop at the new tip height, but that wasn't happening.. So instead this hack is to force stop the sync..  This is feature gated as `test`, and cannot be used in production..

Reply:
Ah gotcha, ok.